### PR TITLE
Fix/4.2.3/stad 172

### PR DESCRIPTION
**Pull request by hb0**

### CODE REVIEW

File: datacapturing/src/androidTest/java/de/cyface/datacapturing/model/CapturedDataWriterTest.java [hunk 1]
```diff
@@ -153,7 +153,7 @@ public void tearDown() {
     /**
      * Tests whether creating and closing a measurement works as expected.
      *
-     * @throws CursorIsNullException If {@link ContentProvider} was inaccessible.
+     * @throws CursorIsNullException      If {@link ContentProvider} was inaccessible.
```
Comment (muthenberg) [inline datacapturing/src/androidTest/java/de/cyface/datacapturing/model/CapturedDataWriterTest.java]: These whitespaces are not from our default formatter.

File: datacapturing/src/androidTest/java/de/cyface/datacapturing/model/CapturedDataWriterTest.java [hunk 2]
```diff
@@ -169,7 +169,7 @@ public void testCreateNewMeasurement() throws NoSuchMeasurementException, Cursor
         Cursor result = null;
         try {
             result = mockResolver.query(getMeasurementUri(AUTHORITY), null, BaseColumns._ID + "=?",
-                    new String[] {identifierString}, null);
+                    new String[]{identifierString}, null);
```
Comment (muthenberg) [inline datacapturing/src/androidTest/java/de/cyface/datacapturing/model/CapturedDataWriterTest.java]: Do you have the correct formatter. I believe this space should remain.

File: datacapturing/src/androidTest/java/de/cyface/datacapturing/model/CapturedDataWriterTest.java [hunk 3]
```diff
@@ -203,7 +203,7 @@ public void testCreateNewMeasurement() throws NoSuchMeasurementException, Cursor
         Cursor finishingResult = null;
         try {
             finishingResult = mockResolver.query(getMeasurementUri(AUTHORITY), null, BaseColumns._ID + "=?",
-                    new String[] {identifierString}, null);
+                    new String[]{identifierString}, null);
```
Comment (muthenberg) [inline datacapturing/src/androidTest/java/de/cyface/datacapturing/model/CapturedDataWriterTest.java]: same

File: datacapturing/src/androidTest/java/de/cyface/datacapturing/model/CapturedDataWriterTest.java [hunk 4]
```diff
@@ -354,7 +354,7 @@ public void writingDataCompleted() {
         try {
             geoLocationsCursor = mockResolver.query(getGeoLocationsUri(AUTHORITY), null,
                     GeoLocationsTable.COLUMN_MEASUREMENT_FK + "=?",
-                    new String[] {Long.toString(measurement.getIdentifier())}, null);
+                    new String[]{Long.toString(measurement.getIdentifier())}, null);
```
Comment (muthenberg) [inline datacapturing/src/androidTest/java/de/cyface/datacapturing/model/CapturedDataWriterTest.java]: same

File: datacapturing/src/androidTest/java/de/cyface/datacapturing/model/CapturedDataWriterTest.java [hunk 5]
```diff
@@ -463,7 +463,7 @@ public void writingDataCompleted() {
         try {
             geoLocationsCursor = mockResolver.query(getGeoLocationsUri(AUTHORITY), null,
                     GeoLocationsTable.COLUMN_MEASUREMENT_FK + "=?",
-                    new String[] {Long.valueOf(measurement.getIdentifier()).toString()}, null);
+                    new String[]{Long.valueOf(measurement.getIdentifier()).toString()}, null);
```
Comment (muthenberg) [inline datacapturing/src/androidTest/java/de/cyface/datacapturing/model/CapturedDataWriterTest.java]: same

File: datacapturing/src/androidTest/java/de/cyface/datacapturing/model/CapturedDataWriterTest.java [hunk 6]
```diff
@@ -477,7 +477,7 @@ public void writingDataCompleted() {
         Cursor eventsCursor = null;
         try {
             eventsCursor = mockResolver.query(getEventUri(AUTHORITY), null, EventTable.COLUMN_MEASUREMENT_FK + "=?",
-                    new String[] {Long.valueOf(measurement.getIdentifier()).toString()}, null);
+                    new String[]{Long.valueOf(measurement.getIdentifier()).toString()}, null);
```
Comment (muthenberg) [inline datacapturing/src/androidTest/java/de/cyface/datacapturing/model/CapturedDataWriterTest.java]: same

File: persistence/src/main/java/de/cyface/persistence/PersistenceLayer.java [hunk 7]
```diff
@@ -772,53 +777,85 @@ private Cursor loadEventsCursor(final long measurementIdentifier) {
...
             if (track.getGeoLocations().size() > 0) {
                 tracks.add(track);
             }
+
+            // Pause reached: Move geoLocationCursor to the first location of the next sub-track
+            // We do this to ignore locations between pause and resume event (STAD-140)
+            moveCursorToFirstAfter(geoLocationCursor, resumeEventTime);
+
+            // Stop GeoLocation collection if we already reached the last GeoLocation
+            if (geoLocationCursor.isAfterLast()) {
```
Comment (muthenberg) [inline persistence/src/main/java/de/cyface/persistence/PersistenceLayer.java]: This is weird. This is the last command in the while loop. You could just "or" connect this within the while condition, couldn't you?

Comment (hb0) [reply persistence/src/main/java/de/cyface/persistence/PersistenceLayer.java]: Done that.

**Pull request merged**